go/adbc/driver/bigquery/statement.go

```diff
@@ -0,0 +1,525 @@
+	case OptionIntQueryJobTimeout:
+		st.query.QueryConfig.JobTimeout = time.Duration(value) * time.Millisecond
+	default:
+		return adbc.Error{
+			Code: adbc.StatusInvalidArgument,
+			Msg:  fmt.Sprintf("unknown statement string type option `%s`", key),
+		}
+	}
+	return nil
+}
+
+// SetSqlQuery sets the query string to be executed.
+//
+// The query can then be executed with any of the Execute methods.
+// For queries expected to be executed repeatedly, Prepare should be
+// called before execution.
+func (st *statement) SetSqlQuery(query string) error {
+	st.query.QueryConfig.Q = query
+	return nil
+}
+
+// ExecuteQuery executes the current query or prepared statement
+// and returns a RecordReader for the results along with the number
+// of rows affected if known, otherwise it will be -1.
+//
+// This invalidates any prior result sets on this statement.
+func (st *statement) ExecuteQuery(ctx context.Context) (array.RecordReader, int64, error) {
+	parameters, err := st.getQueryParameter()
+	if err != nil {
+		return nil, 0, err
+	}
+	if parameters != nil {
+		st.query.QueryConfig.Parameters = parameters
+	}
+	reader, affectedRows, err := newRecordReader(ctx, st.query, st.connectionImpl.Alloc)
+	if err != nil {
+		return nil, -1, err
+	}
+	return reader, affectedRows, nil
+}
+
+// ExecuteUpdate executes a statement that does not generate a result
+// set. It returns the number of rows affected if known, otherwise -1.
+func (st *statement) ExecuteUpdate(ctx context.Context) (int64, error) {
+	return -1, adbc.Error{
+		Code: adbc.StatusNotImplemented,
+		Msg:  "ExecuteUpdate not yet implemented for BigQuery driver",
+	}
+}
+
+// ExecuteSchema gets the schema of the result set of a query without executing it.
+func (st *statement) ExecuteSchema(ctx context.Context) (*arrow.Schema, error) {
+	return nil, adbc.Error{
+		Code: adbc.StatusNotImplemented,
+		Msg:  "ExecuteSchema not yet implemented for BigQuery driver",
+	}
+}
+
+// Prepare turns this statement into a prepared statement to be executed
+// multiple times. This invalidates any prior result sets.
+func (st *statement) Prepare(_ context.Context) error {
+	return nil
+}
+
+// SetSubstraitPlan allows setting a serialized Substrait execution
+// plan into the query or for querying Substrait-related metadata.
+//
+// Drivers are not required to support both SQL and Substrait semantics.
+// If they do, it may be via converting between representations internally.
+//
+// Like SetSqlQuery, after this is called the query can be executed
+// using any of the Execute methods. If the query is expected to be
+// executed repeatedly, Prepare should be called first on the statement.
+func (st *statement) SetSubstraitPlan(plan []byte) error {
+	return adbc.Error{
+		Code: adbc.StatusNotImplemented,
+		Msg:  "Substrait not yet implemented for BigQuery driver",
+	}
+}
+
+func arrowValueToQueryParameterValue(value arrow.Array) bigquery.QueryParameter {
+	parameter := bigquery.QueryParameter{}
```

Comment:
This is only checking the first element of the array, shouldn't this be handling *all* the values? or shouldn't the caller be looping over the input to execute multiple times with the different bound values?